tests/distributed_mode/test_single_label_classifier.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+import syft as sy
+from syft.generic.string import String
+import syfertext
+from syfertext.pointers.doc_pointer import DocPointer
+from syfertext.pipeline.single_label_classifier import AverageDocEncoder, SingleLabelClassifier
+from syfertext.local_pipeline import get_test_language_model
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# Create a torch hook for PySyft
+hook = sy.TorchHook(torch)
+seed = 42
+
+# Set up local and remote workers
+me = hook.local_worker
+me.is_client_worker = False
+
+alice = sy.VirtualWorker(hook, id='alice')
+bob = sy.VirtualWorker(hook, id='bob')
+crypto_provider = sy.VirtualWorker(hook, id='crypto_provider')
+
+# Create toy classifier
+class Net(sy.Plan):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.fc = nn.Linear(300, 2)
+
+    def forward(self, x):
+        logits = self.fc(x)
+        probs = F.relu(logits)
+        return probs, logits
+
+doc_encoder = AverageDocEncoder()
+labels = ["A", "B"]
+
+
+@pytest.mark.parametrize("text,expected", [("The quick brown fox", "B")])
+def test_single_label_inference(text, expected):
+    torch.manual_seed(seed)
+    classifier_name = "classifier"
+    nlp = get_test_language_model()
+
+    # Set up SingleLabelClassifier and add to pipeline
+    net = Net()
+    classifier = SingleLabelClassifier(
+        classifier=net, 
+        doc_encoder=doc_encoder,
+        encryption=None,
+        labels=labels
+    )  
+    nlp.add_pipe(classifier, name=classifier_name, access={'bob'})
+    nlp.deploy(worker=bob)
+
+    string_ptr = String(text).send(bob)
+
+    # Infer using deployed pipeline
+    doc_ptr1 = nlp(string_ptr) 
```

Comment:
This is not a deployed pipeline,why? because you are using the states that are saved to your local worker's store that were created before deployment.

The comment should be `#infer using pipeline before deployment`